pkgs/servers/monitoring/uptime-kuma/default.nix

```diff
@@ -0,0 +1,56 @@
+{ pkgs, lib, stdenv, fetchFromGitHub, fetchzip, substituteAll, nixosTests, iputils }:
+let
+  deps = import ./composition.nix { inherit pkgs; };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "uptime-kuma";
+  version = "1.18.0";
+
+  src = fetchFromGitHub {
+    owner = "louislam";
+    repo = "uptime-kuma";
+    rev = finalAttrs.version;
+    sha256 = "sha256-XcHgwYswkaqKrRWxE2TsLEZHQN535boPMhaLWxV3pn8=";
+  };
+
+  patches = [
+    # Fixes the permissions of the database being not set correctly
+    # See https://github.com/louislam/uptime-kuma/pull/2119
+    ./fix-database-permissions.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace server/ping-lite.js \
+      --replace "/bin/ping" "${iputils}/bin/ping" \
+    --replace "/sbin/ping" "${iputils}/bin/ping" \
+    --replace "/sbin/ping6" "${iputils}/bin/ping"
+  '';
+
+  installPhase = ''
+    mkdir -p $out/
+    cp -r . $out/
+    ln -s ${deps.package}/lib/node_modules/uptime-kuma/node_modules/ $out
+    ln -s ${finalAttrs.passthru.ui} $out/dist
+  '';
+
+  passthru = {
+    tests = {
+      inherit (nixosTests) uptime-kuma;
+    };
+
+    updateScript = ./update.sh;
+
+    ui = fetchzip {
+      name = "uptime-kuma-dist-${finalAttrs.version}";
+      url = "https://github.com/louislam/uptime-kuma/releases/download/${finalAttrs.version}/dist.tar.gz";
+      sha256 = "sha256-2fO3OcabCDjGj7Gnniqf3Mw4t90/05P2ofmweHu0oXY=";
```

Comment:
is it worth having a uiSha256 on the base definition so it could also benefit from `finalAttrs` or nah?

Reply:
I think it is interesting ! I've added that